src/models/transactions/offerCreate.ts

```diff
@@ -0,0 +1,47 @@
+import { ValidationError } from "../../common/errors";
+import { Amount, IssuedCurrencyAmount } from "../common";
+import { CommonFields, verifyCommonFields } from "./common";
+
+export interface OfferCreate extends CommonFields {
+    TransactionType: "OfferCreate";
+    Expiration?: number;
+    OfferSequence?: number;
+    TakerGets: Amount;
+    TakerPays: Amount;
+}
+
+/**
+ * Verify the form and type of an OfferCreate at runtime.
+ * 
+ * @param tx - An OfferCreate Transaction
+ * @returns - Void.
+ * @throws - When the OfferCreate is Malformed.
+ */
+ export function verifyOfferCreate(tx: OfferCreate): void {
```

Comment:
I imagine we may want to add some preflight checks here in the future. (or have we decided against that?) If so, we should make this `async` and return a Promise in case we need to check the state of the ledger.

Reply:
Well, I guess this isn't part of a client, so I'm unsure what value this `verify` actually adds. The most valuable/tricky preflight checks are ones that involve checking a few things about the ledger to see if we can prevent an obvious `tec`. Maybe it would be like `client.verify(transaction)`

Reply:
Yeah, that's a really good idea.

Reply:
right now, verify just checks transaction form at runtime, essentially replacing `jsonschemas`

Reply:
I was hoping we could get rid of jsonschemas and just not check transaction form anymore. I don't think checking for missing fields, etc. adds much value in practice. Transaction form is generally not problematic, types already provide some safety, and it is ultimately verified by rippled anyway. I think it's best to let rippled do it since that is guaranteed to be 100% correct and never needs to be updated (i.e. future versions of rippled could add new fields or otherwise change the definition of what is considered well-formed).

Reply:
I think we should still be doing some runtime checks for transactions, because the potential cost of screwing something up is high. For example, we could add the `AccountDelete` checks in this type of method. Anything that requires checking `rippled` can be done in `autofill` or something, but IMO the verify methods shouldn't require a client.

Reply:
The important `AccountDelete` checks require a client to do correctly. Note that the owner reserve may change in the future, so I think we should avoid hardcoding it. One way to get it is `server_info` -> `reserve_inc_xrp`